[Feat] 카테고리, 액션 필요/불필요의 2중 분류 적용

canolayoo78 opened the pull request:
## 📝 Summary

기존에 존재하던 카테고리 분류 외에
메일을 받은 수신인이 액션이 필요한지, 불필요한지를 분류하는 신규 분류를 추가합니다.

<!--
PR의 주요 내용을 간단히 요약해주세요.
예: 로그인 기능에서 사용자 세션 관리 개선.
-->

## ✅ Checklist

<!--
해당되는 항목을 [x]로 만들어주세요.
예: - [x] 관련 이슈가 명시되어 있습니다.

해당되지 않는 항목은 앞뒤로 ~를 붙여서 취소선을 그어주세요.
예: - ~[ ] 관련 이슈가 명시되어 있습니다.~
-->

~- [ ] 관련 이슈가 명시되어 있습니다.~
- [x] 테스트가 완료되었습니다.
~- [ ] 문서 업데이트가 포함되었습니다.~
- [x] 코드 리뷰를 위한 사전 검토를 완료했습니다.

## 📄 Description
<img width="908" alt="image" src="https://github.com/user-attachments/assets/c8e11ccd-99b0-4397-ba83-8d4f2a50c123" />

- 고정되어있던 분류 기준을 다양하게 변경 가능하도록 수정하였습니다.
   현재는 2가지 분류 기준을 적용하여 분류를 실행합니다.
- 다중 분류에 맞는 map_category 함수를 작성하였습니다.

<!--
PR의 변경 사항을 구체적으로 설명해주세요.
예: 이번 변경 사항에는 로그인 세션 시간 연장 및 오류 메시지 개선이 포함됩니다.
-->

## 💡 Notice (Optional)

실제 서비스 환경을 위해
Gmail fetch 시, 전일 9시 이후에 수신한 메일만 처리하도록 하는 분기가 추가되었습니다.
테스트 환경에서는 수신일시와 관계없이 처리가 가능하도록 pass 처리하여
fetch 함수 구동 결과에는 변화가 없습니다.

<!--
리뷰어가 알아야 할 추가적인 사항이나 주의점이 있다면 작성해주세요.
예: 이 기능은 인증 모듈과 호환성을 고려해야 합니다.
-->

## 🔗 Related Issue(s)

<!--
이 PR과 관련된 이슈나 기존 PR이 있다면 번호를 언급하고, 필요 시 close할 이슈도 명시해주세요.
예: 관련 이슈 - #33, close #27
-->

gsgh3016 reviewed (changes requested):
`Mail.similar_mails` 변수를 삭제해서 클러스터링 동작이 중간에 멈추는 것으로 확인됩니다. 해당 내용 검토 부탁드려요!

canolayoo78 committed:
feat: `main.py`에 다중 분류 추가

다중 Mail 객체 dict를 통한 refine / summary의 경우 고려되지 않습니다.
generate_report() 함수에 LLM을 활용하지 않는 수정 전까지 오류가 발생합니다.

canolayoo78 committed:
feat: classification 기준 동적 설정

- 전역변수로 고정되어있던 classification 기준 yaml file을
  classificaiton_type에 맞춰 동적 변경

- 해당 변경과 함께 refine 및 final report 생성 시 error 발생으로 main 주석처리

canolayoo78 committed:
feat: Gmail fetch 시, 전일 9시 이후에 수신한 메일만 처리

테스트 환경에서는 수신일시와 관계없이 처리가 가능하도록 pass 처리함

canolayoo78 committed:
feat: 다중 분류에 맞게 map_category 함수 작성

canolayoo78 committed:
fix: 불필요하게 추가된 `time.sleep()` 제거
